### PR TITLE
[MINOR] Switch to FileIO for InputFormat

### PR DESCRIPTION
**Pull request by holdenk**
There was a TODO saying to switch to FileIO from directly accessing the HadoopInputFile class.

### CODE REVIEW

**Comment by rdblue** [comment]
Looks good, and helps unblock #1497. Thanks @holdenk!

File: mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java [hunk 1]
```diff
@@ -165,13 +166,15 @@ private static void checkResiduals(CombinedScanTask task) {
...
     private CloseableIterator<T> currentIterator;
+    private FileIO fileIO;
 
     @Override
     public void initialize(InputSplit split, TaskAttemptContext newContext) {
       Configuration conf = newContext.getConfiguration();
       // For now IcebergInputFormat does its own split planning and does not accept FileSplit instances
       CombinedScanTask task = ((IcebergSplit) split).task();
       this.context = newContext;
+      this.fileIO = new HadoopFileIO(this.context.getConfiguration());
```
Comment (rdblue) [inline mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java:177]: Nit: we use `this.` to distinguish instance fields when setting them, but there's no need when accessing them.

Comment (holdenk) [reply mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java:177]: thanks, I'll keep that in mind :)

Comment (kbendick) [reply mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java:177]: Good to know! Do we document these style preferences anywhere (using `this`, when we use the term `project` vs `select`, etc)? @rdblue

**Pull request merged**